src/main/java/com/drullkus/thermalsmeltery/ThermalSmeltery.java

```diff
@@ -22,6 +22,9 @@
 @Mod(modid = LibMisc.MOD_ID, name = LibMisc.MOD_NAME, dependencies = LibMisc.DEPENDENCIES)
 public class ThermalSmeltery
 {
+    @Mod.Instance(value = LibMisc.MOD_ID)
+    public static ThermalSmeltery instance = new ThermalSmeltery();
```

Comment:
I double checked how FML works, and it actually forcefully sets the instance field no matter if it is instantiated already or not. So while your method _does_ work, you are creating an instance of the Mod class that is never used.